test/mix/tasks/phx.new_test.exs

```diff
@@ -1,100 +1,102 @@
-# Remove archive to avoid conflicts
-with list when is_list(list) <- :code.lib_dir(:phx_new) do
-  :code.del_path(:filename.join(list, 'ebin'))
-end
-
-# Mock live reloading for testing the generated application.
-defmodule Phoenix.LiveReloader do
-  def init(opts), do: opts
-  def call(conn, _), do: conn
-end
-
-_ = Kernel.ParallelCompiler.compile Path.wildcard("installer/lib/**/*.ex")
-Code.require_file("installer/test/mix_helper.exs")
-
-# Define a fake live reload socket.
-defmodule Phoenix.LiveReloader.Socket do
-  def child_spec(_) do
-    Supervisor.child_spec({Task, fn -> :ok end}, [])
+if Version.match?(System.version(), ">= 1.11.0") do
```

Comment:
Maybe this could be inverted to `Mix.raise` if the version was smaller than 1.11. It would make the diff cleaner.

```elixir
if Version.match?(System.version(), ">= 1.11.0") do
  Mix.raise(...)
end

# Remove archive to avoid conflicts
with list when is_list(list) <- :code.lib_dir(:phx_new) do
```

Reply:
If we raise the tests will fail. It is fine to leave this as is for now because we plan to move all of this elsewhere soon!